### ISSUE TITLE
How create where in joined objects?

### ISSUE THREAD

**Issue by iformas** [simolus3/moor#252]
Hi, I want to create a where query for joined objects but i can't find examples for this approach
I have the following table structure:
Region->Provincia->Comuna

I want to select the "Region" with x "Comuna".. something like:
```dart
..where((t) => t.Provincia.Comuna.comCodNac.equals(id))).getSingle()
```

but where only have access to "Region"

Here is a query example:
```dart
 Future<Region> getRegionByComuna(int id) =>
      (select(regions)..join(
          [
            leftOuterJoin(provincias,provincias.prvRegCod.equalsExp(regions.regCod)),
            leftOuterJoin(comunas,comunas.comProvinciaCod.equalsExp(provincias.prvCodNac)),
          ])
..where((t) =>?????).getSingle();
```

what should be the right way to solve this?

**Comment by simolus3** [comment]
TLDR: This might just be a typo. The first line should be `(select(regions).join(` instead of `(select(regions)..join(`

In Dart, the `..` operator can be used to chain multiple calls on the same object, see the [introduction here](https://news.dartlang.org/2012/02/method-cascades-in-dart-posted-by-gilad.html). In moor, calling `.join` turns a simple select statement into a more complicated joined select. The reason for being more complicated is that the join could be on any number of tables. So when you're writing `..join`, you're creating the joined select statement but then continue to use the simple select statement, where the joined tables aren't available.

Something like the following should work.
```dart
 Future<Region> getRegionByComuna(int id) =>
      (select(regions).join(
          [
            leftOuterJoin(provincias, provincias.prvRegCod.equalsExp(regions.regCod)),
            leftOuterJoin(comunas, comunas.comProvinciaCod.equalsExp(provincias.prvCodNac)),
          ])
..where(communas.someId.equals(id)).getSingle();
```
Personally I prefer to split complicated queries by introducing a local variable. That can make the code easier to understand and works better when using the formatter:
```dart
final query = select(regions).join([
  leftOuterJoin(provincias, provincias.prvRegCod.equalsExp(regions.regCod)),
  leftOuterJoin(comunas, comunas.comProvinciaCod.equalsExp(provincias.prvCodNac)),
])..where(communas.someId.equals(id));

return query.getSingle();
```

Let me know if you run into any problems with that.

**Comment by iformas** [comment]
Thanks for the explanation and the example!.. it works like a charm.
